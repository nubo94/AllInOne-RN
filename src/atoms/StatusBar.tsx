import React from 'react';
import {StatusBar as SB, StatusBarProps, useColorScheme} from 'react-native';

interface IStatusBarProps extends StatusBarProps {}

const StatusBar = ({...props}: IStatusBarProps) => {
  const isDarkMode = useColorScheme() === 'dark';

  return (
    <SB barStyle={isDarkMode ? 'light-content' : 'dark-content'} {...props} />
  );
};

export default StatusBar;
