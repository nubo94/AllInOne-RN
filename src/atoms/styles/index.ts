import * as Colors from './colors';
import * as Sizing from './sizing';
import * as Outlines from './outlines';
import * as Typography from './typography';

export {Colors, Sizing, Typography, Outlines};
