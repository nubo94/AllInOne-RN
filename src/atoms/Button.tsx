import React from 'react';
import {StyleSheet} from 'react-native';
import {
  TouchableOpacity,
  ActivityIndicator,
  TouchableOpacityProps,
} from 'react-native';

// Custom Component
import Text from './Text';

// Theme
import {Outlines, Colors, Sizing, Typography} from './styles';
export interface IButtonProps extends TouchableOpacityProps {
  title: string;
  disabled?: boolean;
}

const Button = ({disabled, style, title, ...props}: IButtonProps) => {
  return (
    <TouchableOpacity
      activeOpacity={0.8}
      disabled={disabled}
      style={[style, disabled ? disabledStyle.main : classes.main]}
      {...props}>
      {disabled ? (
        <ActivityIndicator size="small" color={Colors.neutral.black} />
      ) : (
        <Text label={title} style={classes.text} />
      )}
    </TouchableOpacity>
  );
};

const classes = StyleSheet.create({
  main: {
    width: '100%',
    display: 'flex',
    height: Sizing.x50,
    padding: Sizing.x10,
    alignItems: 'center',
    maxHeight: Sizing.x50,
    justifyContent: 'center',
    borderColor: Colors.neutral.s150,
    backgroundColor: Colors.primary.brand,
    borderWidth: Outlines.borderWidth.thin,
    borderRadius: Outlines.borderRadius.small,
    ...Outlines.shadow.base,
  },
  text: {
    textAlign: 'center',
    textTransform: 'uppercase',
    color: Colors.neutral.white,
    fontSize: Typography.fontSize.x20.fontSize,
    fontWeight: Typography.fontWeight.bold.fontWeight,
  },
});

const disabledStyle = StyleSheet.create({
  main: {
    ...classes.main,
    backgroundColor: Colors.neutral.s250,
    ...Outlines.shadow.disable,
  },
});

export default Button;
