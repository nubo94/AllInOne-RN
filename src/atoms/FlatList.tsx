import React from 'react';
import {FlatList as Fl, FlatListProps, StyleSheet} from 'react-native';
import {Text, View} from '.';
import {Colors, Typography} from './styles';

interface IFlatListProps extends FlatListProps<any> {
  empty: string;
}

const FlatList = ({empty, ...props}: IFlatListProps) => {
  return (
    <Fl
      ListEmptyComponent={
        <View style={classes.main}>
          <Text style={classes.empty} label={empty} />
        </View>
      }
      {...props}
    />
  );
};

const classes = StyleSheet.create({
  main: {
    display: 'flex',
    alignItems: 'center',
  },
  empty: {
    color: Colors.neutral.s400,
    fontSize: Typography.fontSize.x10.fontSize,
  },
});

export default FlatList;
