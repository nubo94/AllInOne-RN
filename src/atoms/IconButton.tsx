import React from 'react';
import Icon from 'react-native-vector-icons/Feather';
import {
  StyleSheet,
  useColorScheme,
  TouchableOpacity,
  ActivityIndicator,
  TouchableOpacityProps,
} from 'react-native';

// Theme
import {Colors, Sizing} from './styles';
export interface IconButtonProps extends TouchableOpacityProps {
  iconName: string;
  disabled?: boolean;
  colorIconType: 'danger' | 'normal' | 'disable' | 'primary';
}

const IconButton = ({
  iconName,
  disabled,
  colorIconType,
  ...props
}: IconButtonProps) => {
  const isDarkMode = useColorScheme() === 'dark';

  function _color() {
    switch (colorIconType) {
      case 'danger':
        return isDarkMode ? dClasses.danger : lClasses.danger;
      case 'normal':
        return isDarkMode ? dClasses.normal : lClasses.normal;
      case 'disable':
        return isDarkMode ? dClasses.disable : lClasses.disable;
      case 'primary':
        return isDarkMode ? dClasses.primary : lClasses.primary;
      default:
        break;
    }
  }

  return (
    <TouchableOpacity activeOpacity={0.8} disabled={disabled} {...props}>
      {disabled ? (
        <ActivityIndicator
          size="small"
          color={isDarkMode ? Colors.neutral.s400 : Colors.neutral.black}
        />
      ) : (
        <Icon name={iconName} size={Sizing.x20} style={_color()} />
      )}
    </TouchableOpacity>
  );
};

const lClasses = StyleSheet.create({
  normal: {
    color: Colors.neutral.s900,
  },
  danger: {
    color: Colors.danger.s400,
  },
  disable: {
    color: Colors.neutral.s150,
  },
  primary: {
    color: Colors.primary.brand,
  },
});
const dClasses = StyleSheet.create({
  normal: {
    color: Colors.neutral.s300,
  },
  danger: {
    color: Colors.danger.s400,
  },
  disable: {
    color: Colors.neutral.s500,
  },
  primary: {
    color: Colors.neutral.white,
  },
});

export default IconButton;
