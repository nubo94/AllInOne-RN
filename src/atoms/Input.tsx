import React from 'react';
import {StyleSheet} from 'react-native';
import {TextInput as TI, TextInputProps, useColorScheme} from 'react-native';

// Theme
import {Outlines, Colors, Sizing} from './styles';

export interface InputProps extends TextInputProps {}

const Input = ({style, ...props}: InputProps) => {
  const isDarkMode = useColorScheme() === 'dark';

  return (
    <TI
      style={[isDarkMode ? dClasses.main : lClasses.main, style]}
      placeholderTextColor={
        isDarkMode ? Colors.neutral.s500 : Colors.neutral.s250
      }
      {...props}
    />
  );
};

const lClasses = StyleSheet.create({
  main: {
    height: Sizing.x50,
    padding: Sizing.x10,
    maxHeight: Sizing.x50,
    color: Colors.neutral.s900,
    borderColor: Colors.neutral.s150,
    backgroundColor: Colors.neutral.white,
    borderWidth: Outlines.borderWidth.thin,
    borderRadius: Outlines.borderRadius.small,
    ...Outlines.shadow.base,
  },
});

const dClasses = StyleSheet.create({
  main: {
    ...lClasses.main,
    color: Colors.neutral.s150,
    borderColor: Colors.neutral.s500,
    shadowColor: Colors.neutral.black,
    backgroundColor: Colors.neutral.s900,
  },
});

export default Input;
