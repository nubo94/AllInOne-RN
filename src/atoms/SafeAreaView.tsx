import React from 'react';
import {SafeAreaView as SAV, useColorScheme} from 'react-native';

// Theme
import {Colors} from './styles';

interface ISafeAreaView {
  children: JSX.Element[] | JSX.Element;
}

const SafeAreaView = ({children, ...props}: ISafeAreaView) => {
  const isDarkMode = useColorScheme() === 'dark';
  const backgroundStyle = {
    backgroundColor: isDarkMode
      ? Colors.darker.primaryBackground.main
      : Colors.lighter.primaryBackground.main,
  };
  return (
    <SAV style={backgroundStyle} {...props}>
      {children}
    </SAV>
  );
};

export default SafeAreaView;
