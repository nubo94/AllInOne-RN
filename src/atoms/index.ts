import SafeAreaView from './SafeAreaView';
import Text from './Text';
import View from './View';
import StatusBar from './StatusBar';
export {Text, View, StatusBar, SafeAreaView};
