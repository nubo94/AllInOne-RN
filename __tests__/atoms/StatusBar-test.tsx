/**
 * Describe component StatusBar
 * @format
 */

import React from 'react';
import renderer from 'react-test-renderer';
import {StatusBar} from '../../src/atoms';

test('renders correctly - StatusBar', () => {
  const tree = renderer.create(<StatusBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
