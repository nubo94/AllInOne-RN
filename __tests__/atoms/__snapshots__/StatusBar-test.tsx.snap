// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - StatusBar 1`] = `null`;
